lib/hanami/commands/generate/action.rb

```diff
@@ -234,15 +235,17 @@ def spec_root
         end
 
         def relative_action_path
-          path = Pathname.new('.').join('..', '..', '..')
-          path = path.join('..') if environment.container?
-          path.join(application_path, 'controllers', @controller_pathname, @action_name.downcase)
+          relative_base_path.join(application_path, 'controllers', *@controller_patharray, @action_name.downcase)
         end
 
         def relative_view_path
-          path = Pathname.new('.').join('..', '..', '..')
-          path = path.join('..') if environment.container?
-          path.join(application_path, 'views', @controller_pathname, @action_name.downcase)
+          relative_base_path.join(application_path, 'views', *@controller_patharray, @action_name.downcase)
+        end
+
+        def relative_base_path
+          nestings = ['..'] * @controller_pathname.count("/")
```

Comment:
@berfarah Thanks for the clarification. Having these slashes around is confusing, that's why we need to name them via constants.

To avoid further ping/pong, I'm gonna merge this and cleanup by myself.